hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -0,0 +1,301 @@
+import com.hedera.node.app.service.token.ReadableAccountStore;
+import com.hedera.node.app.services.ServiceScopeLookup;
+import com.hedera.node.app.signature.ExpandedSignaturePair;
+import com.hedera.node.app.signature.SignatureExpander;
+import com.hedera.node.app.signature.SignatureVerificationFuture;
+import com.hedera.node.app.signature.SignatureVerifier;
+import com.hedera.node.app.spi.info.NodeInfo;
+import com.hedera.node.app.spi.signatures.SignatureVerification;
+import com.hedera.node.app.spi.workflows.HandleContext.TransactionCategory;
+import com.hedera.node.app.spi.workflows.HandleException;
+import com.hedera.node.app.spi.workflows.PreCheckException;
+import com.hedera.node.app.state.HederaState;
+import com.hedera.node.app.workflows.TransactionChecker;
+import com.hedera.node.app.workflows.dispatcher.ReadableStoreFactory;
+import com.hedera.node.app.workflows.dispatcher.TransactionDispatcher;
+import com.hedera.node.app.workflows.prehandle.PreHandleContextImpl;
+import com.hedera.node.app.workflows.prehandle.PreHandleResult;
+import com.hedera.node.app.workflows.prehandle.PreHandleResult.Status;
+import com.hedera.node.app.workflows.prehandle.PreHandleWorkflow;
+import com.hedera.node.config.ConfigProvider;
+import com.swirlds.common.system.Round;
+import com.swirlds.common.system.events.ConsensusEvent;
+import com.swirlds.common.system.transaction.ConsensusTransaction;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import javax.inject.Inject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * The handle workflow that is responsible for handling the next {@link Round} of transactions.
+ */
+public class HandleWorkflow {
+
+    private static final Logger LOG = LogManager.getLogger(HandleWorkflow.class);
+
+    private final NodeInfo nodeInfo;
+    private final PreHandleWorkflow preHandleWorkflow;
+    private final TransactionDispatcher dispatcher;
+    private final RecordManager recordManager;
+    private final SignatureExpander signatureExpander;
+    private final SignatureVerifier signatureVerifier;
+    private final TransactionChecker checker;
+    private final ServiceScopeLookup serviceScopeLookup;
+    private final ConfigProvider configProvider;
+
+    @Inject
+    public HandleWorkflow(
+            @NonNull final NodeInfo nodeInfo,
+            @NonNull final PreHandleWorkflow preHandleWorkflow,
+            @NonNull final TransactionDispatcher dispatcher,
+            @NonNull final RecordManager recordManager,
+            @NonNull final SignatureExpander signatureExpander,
+            @NonNull final SignatureVerifier signatureVerifier,
+            @NonNull final TransactionChecker checker,
+            @NonNull final ServiceScopeLookup serviceScopeLookup,
+            @NonNull final ConfigProvider configProvider) {
```

Comment:
After #6754 is merged, we will need to add the `HederaRecordCache` as a dependency, and use it to check for and handle duplicates.

Reply:
Create #6765 to track this.